plugins/outputs/cloudwatchlogs/internal/pusher/target.go

```diff
@@ -135,17 +146,17 @@ func (m *targetManager) createLogStream(t Target) error {
 
 func (m *targetManager) processRetentionUpdates() {
 	for target := range m.retentionPolicyUpdates {
-		err := m.processWithRetry(target)
+		err := m.processWithRetry(target, false)
 		if err != nil {
 			m.logger.Errorf("Failed to update retention policy for target %v: %v", target, err)
 		}
 	}
 }
 
-func (m *targetManager) processWithRetry(target Target) error {
+func (m *targetManager) processWithRetry(target Target, newGroup bool) error {
 	var err error
 	for attempt := 0; attempt < maxAttempts; attempt++ {
-		err = m.checkAndUpdate(target)
+		err = m.checkAndUpdate(target, newGroup)
```

Comment:
nit: Would expect to not call `checkAndUpdate` if `newGroup` is true.
```suggestion
        if newGroup {
          err = m.updateRetentionPolicy(target)
        } else {
		  err = m.checkAndUpdate(target)
		}
```

Reply:
`checkAndUpdate` will handle that